src/mpid/common/shm/mpidu_init_shm.c

```diff
@@ -15,6 +15,14 @@ typedef struct Init_shm_barrier {
     MPL_atomic_int_t wait;
 } Init_shm_barrier_t;
 
+/* The first cacheline of the shm block is reserved for synchronization purpose */
+typedef struct Init_shm_header {
+    Init_shm_barrier_t barrier;
+#ifdef MPL_USE_LOCK_BASED_PRIMITIEVS
+    MPL_emulation_ipl_t atomic_lock;
```

Comment:
If I am not mistaken previous OPA IPL code was placing the lock variable in a separate cacheline (with respect to the barrier). What would be the implication of having these in the same cacheline in this code?

Reply:
Hmm... The lock itself is not an atomic variable, so I don't see any problems.

Reply:
Well, I take it back. A mutex internally must be an atomic variable. So there potentially is the false sharing concerns, that one process updates the barrier counter will force all process/thread that using atomics to reload the cacheline. It remains to see if this is a concern. When all processes are in the barrier, the cacheline need to be reloaded anyway.

Reply:
The lock is acquired before updating the barrier variable right? So having them in the same cache line should actually be faster?

Reply:
I am not sure about the "faster" part, but probably not of much concern. :)

Reply:
I see, yes there is false sharing but as you said when processes are in the barrier cache line has to be bounced between caches anyway.